dpypelines/pipeline/generic_file_ingress_v1.py

```
@@ -0,0 +1,184 @@
+                        "required_file": required_file,
+                    },
+                )
+                de_notifier.failure()
+        except Exception as err:
+            files_in_directory = local_store.get_file_names()
+            logger.error(
+                "Error occurred when looking for required file",
+                err,
+                data={
+                    "required_file": required_file,
+                    "required_file_patterns": required_file_patterns,
+                    "files_in_directory": files_in_directory,
+                    "pipeline_config": pipeline_config,
+                },
+            )
+            de_notifier.failure()
+            raise err
+
+    # Allow DE's to skip the upload to s3 part of the pipeline while
+    # developing code locally.
+    skip_data_upload = os.environ.get("SKIP_DATA_UPLOAD", False)
+    if skip_data_upload is not False:
+        try:
+            skip_data_upload = str_to_bool(skip_data_upload)
+        except Exception as err:
+            logger.error(
+                "Unable to cast SKIP_DATA_UPLOAD to boolean",
+                err,
+                data={"value": skip_data_upload},
+            )
+            de_notifier.failure()
+            raise err
+
+    logger.info(
+        "skip_data_upload set from SKIP_DATA_UPLOAD env var",
+        data={"value": skip_data_upload},
+    )
+
+    if skip_data_upload is not True:
+
+        # Upload output files to Upload Service
+        try:
+            # Create UploadClient from upload_url
+            UploadServiceClient(upload_url)
+            logger.info(
+                "UploadClient created from upload_url", data={"upload_url": upload_url}
+            )
+        except Exception as err:
+            logger.error(
+                "Error creating UploadClient", err, data={"upload_url": upload_url}
+            )
+            de_notifier.failure()
+            raise err
+
```

the above block just creates `upload_service_client` (with the tweak I've put in a comment).

now you have an instantiated upload_client you need to:

- loop through all the require files
- ignore/continue if the file iis called "manifest .json"
- otherwise upload it with the upload client. 
       - if its an xml file, then you'd need to use `upload_client.upload_new_sdmx(<path to file>)`
       - if its a csv file, then you'd need to use `upload_client.upload_new_csv(<path to file>)`
       - any other type then (for now) you;'ll need to raise an error as these are the currently supported file types. 
       
theres some documentation on the uplaod client here: https://github.com/ONSdigital/dp-python-tools/tree/develop/dpytools/http. You onyl want the methods with "new" in them.

---

Thanks; done